src/main/java/com/example/demo/serviceIml/StudentServiceImpl.java

```diff
@@ -44,37 +42,30 @@ public void addNewStudent(Student student) {
 
 	@Transactional
 	@Override
-	public void updateStudent(int studentId, String studentName, String studentEmail, LocalDate studentDOB) {
-		Student student = studentRepository.findById(studentId)
+	public void updateStudent(int studentId, Student updateStudent) {
+		Student currentStudent = studentRepository.findById(studentId)
 				.orElseThrow(() -> new NotFoundException(studentNotExistByIdMsg + studentId));
 
 		// Student name processing
-		if (studentName == null || studentName.length() < 2 || Objects.equals(student.getStudentName(), studentName)) {
+		if (updateStudent.getStudentName().length() < 2) {
 			throw new BadRequestException(CommonResponses.nameNotValidMsg);
```

Comment:
Here is returning `Name is not valid. Name can not be null or single character` This is just checking if the name is a single character. Change output message for only a single character